ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -55,4 +55,29 @@ - (void)testShouldIgnoreVnodeType {
     XCTAssertFalse(ShouldIgnoreVnodeType(static_cast<vtype>(1000), testVnode.get()));
 }
 
+- (void)testFileFlaggedInRoot {
+    bool fileFlaggedInRoot;
+    MountPointer testMount = mount::Create("hfs", fsid_t{}, 0);
+    VnodePointer testVnode = vnode::Create(testMount, "/foo");
+    
+    testVnode->SetAttr(FileFlags_IsInVirtualizationRoot);
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertTrue(fileFlaggedInRoot);
+    
+    testVnode->SetAttr(FileFlags_IsEmpty);
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->SetAttr(FileFlags_Invalid);
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->SetAttr(100);
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->SetGetAttrReturnCode(-1);
```

Comment:
Note that `vnode_getattr` returns errno codes [such as `EBADF`](https://github.com/Microsoft/VFSForGit/issues/328#issuecomment-453285966). I don't think it can ever return a negative value.